services-custom/s3-transfermanager/src/main/java/software/amazon/awssdk/custom/s3/transfer/S3TransferManager.java

```diff
@@ -135,11 +135,7 @@ static S3TransferManager create() {
 
         Builder s3ClientConfiguration(S3ClientConfiguration configuration);
 
-        default Builder s3ClientConfiguration(Consumer<S3ClientConfiguration.Builder> builderConsumer) {
-            S3ClientConfiguration.Builder builder = S3ClientConfiguration.builder();
-            builderConsumer.accept(builder);
-            return this;
-        }
+        Builder s3ClientConfiguration(Consumer<S3ClientConfiguration.Builder> builderConsumer);
```

Comment:
Yeah, the consumer builder method should be in the impl because otherwise `Default#s3ClientConfiguration` will be null and NPE will be thrown. (I should've created a separate PR for this change)

Reply:
Couldn't we just call s3ClientConfiguration(builder.build())? That's what we usually do with other consumer-builder methods.

Reply:
😅 D'oh!